web/base_viewer.js

```diff
@@ -1517,6 +1517,9 @@ class BaseViewer {
     const { eventBus } = this;
 
     const dispatchPageClose = pageNumber => {
+      if (this._pageOpenPendingSet?.has(pageNumber)) {
```

Comment:
The patch looks good.
An unrelated question: why do you lazily initialize `_pageOpenPendingSet` ?
Setting it in `_initializeScriptingEvent` will avoid few checks but maybe it's pretty rare to have a non null `_pageOpenPendingSet`.

Reply:
> but maybe it's pretty rare to have a non null `_pageOpenPendingSet`.

That was my idea when writing the code, since that's what I observed in practice early during development.

However, to initialize things correctly in the general case, the final code also dispatches a "pageopen" event on the pre-existing "pagesinit" event and with those changes `_pageOpenPendingSet` is now always created. While the extra checks probably don't matter much, I'll fix this is a separate commit; thank you!